### PR TITLE
complete testing done

### PR DESCRIPTION
**Pull request by balani98**

### CODE REVIEW

File: client/src/app/auth.guard.ts [hunk 1]
```diff
@@ -2,28 +2,40 @@ import { jsonDecoder } from 'src/app/utils/json.util';
...
+      const now = Date.now().valueOf() / 1000
+         
+      
+       if(token!=null){
+         if(jsonDecoder().role==expectedRole)
+             return true
+          else
+          this.router.navigate(['/404']);
+       }
+       else{
```
Comment (KritikaSachdeva) [inline client/src/app/auth.guard.ts]: two else conditions??

File: client/src/app/auth.guard.ts [hunk 2]
```diff
@@ -2,28 +2,40 @@ import { jsonDecoder } from 'src/app/utils/json.util';
...
      // this will be passed from the route config
     // on the data property
+    const token=localStorage.getItem('Authorization');
+    
     const expectedRole = next.data.expectedRole;
-    const decodeToken = jsonDecoder();
+    console.log(token);
+   
+    // decode the token to get its payload
+
```
Comment (KritikaSachdeva) [inline client/src/app/auth.guard.ts]: remove console.log, blank lines and commented code

File: client/src/app/auth.guard.ts (hunk 1 above)
Comment (balani98) [reply client/src/app/auth.guard.ts]: abhi ke liye yhi idea hai

File: client/src/app/utils/json.util.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (KritikaSachdeva) [inline client/src/app/utils/json.util.ts]: remove these extra blank lines.

File: client/src/main.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import 'hammerjs';
```
Comment (KritikaSachdeva) [inline client/src/main.ts]: remove this import

File: client/src/app/auth.guard.ts (hunk 2 above)
Comment (balani98) [reply client/src/app/auth.guard.ts]: done

**Review by KritikaSachdeva — APPROVED** [review]

**Pull request merged**